src/components/Task.vue

```diff
@@ -0,0 +1,82 @@
+<template>
+  <li :class="{done: task.taskState}">
+    <div>
+      <input
+        type="checkbox"
+        v-model="task.taskState"
```

Comment:
You don't need to modify incoming props in child components, please use events for this. https://vuejs.org/v2/guide/components-props.html#One-Way-Data-Flow